jupyterlab_gcedetails/src/components/server_wrapper.tsx

```diff
@@ -0,0 +1,30 @@
+import { ServerConnection } from '@jupyterlab/services';
+
+export class ServerWrapper {
+  private readonly serverSettings: ServerConnection.ISettings;
+  private readonly url: string;
+  constructor(
+    endpoint: string,
+    serverSettings: ServerConnection.ISettings = ServerConnection.defaultSettings
+  ) {
+    this.serverSettings = serverSettings;
+    this.url = this.serverSettings.baseUrl + endpoint;
+  }
+
+  async get() {
```

Comment:
I would suggest making this interface more explicit. What kind of data is actually being returned? Maybe something like `getUtilizationData`. I would also suggest that the return type be `Promise<Utilization>`. Then, you can just behave like a normal promise and reject if the method is unable to fetch data, rather than return the object with the `ok` property.